### DISCUSSION TITLE
SQL variable

### DISCUSSION THREAD

**Discussion by rydeng** [duckdb/duckdb#5902] [Ideas]
As the title, can duckdb support SQL variable or there is a way to save a value from a query so I can refer to it in a subsequent query?

**Comment by Mytherin** [comment]
Could you be more specific? You could store it in a table, no?

**Reply by rydeng** [reply]
@Mytherin  Thanks for your reply. I want use like this:
`SELECT @id := cust_id FROM customers WHERE cust_id='customer name';
DELETE FROM customers WHERE cust_id = @id;`

**Reply by rydeng** [reply]
or like this.
`SELECT list_transform(['a_', 'b_', 'c_'], duck -> CONCAT(duck, @id))`

**Reply by Mytherin** [reply]
You could use a table for these cases:

```sql
CREATE TABLE id AS SELECT cust_id FROM customers WHERE cust_id='customer name';
DELETE FROM customers WHERE cust_id=(SELECT cust_id FROM id);
```

Although it is of course not as syntactically nice as variables.

Is there another system that this syntax is inspired from?

**Reply by djouallah** [reply]
https://learn.microsoft.com/en-us/sql/relational-databases/stored-procedures/parameters?view=sql-server-ver16

**Reply by rydeng** [reply]
mysql also supports this syntax, https://dev.mysql.com/doc/refman/8.0/en/user-variables.html

**Reply by rydeng** [reply]
Yes, using a table can also achieve this goal.

**Reply by kylelundstedt** [reply]
Here's Snowflake's doc on SQL variables.
https://docs.snowflake.com/en/sql-reference/session-variables.html

**Comment by rydeng** [comment]
@Mytherin But if the value is a list, and I only need an element from list in one select(like list[1], list[2], list[3]...), then I must select from table in every select.
